{{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_widget.py

```diff
@@ -1,28 +1,125 @@
 """
-This module is an example of a barebones QWidget plugin for napari
+This module contains four napari widgets declared in
+different ways:
 
-It implements the Widget specification.
-see: https://napari.org/stable/plugins/guides.html?#widgets
+- a pure Python function flagged with `autogenerate: true`
+    in the plugin manifest. Type annotations are used by
+    magicgui to generate widgets for each parameter. Best
+    suited for simple processing tasks - usually taking
+    in and/or returning a layer.
+- a `magic_factory` decorated function. The `magic_factory`
+    decorator allows us to customize aspects of the resulting
+    GUI, including the widgets associated with each parameter.
+- a `magicgui.widgets.Container` subclass. This provides lots
```

Comment:
It would be good to add a line to this bullet point describing what kind of situations this is best suited for, eg: "Container widgets are most useful when you want to group multiple widgets together for more complex tasks and cannot achieve this with `magicgui_factory`."

Basically, you've done a great job explaining what the four demo examples are, but it would help if we add information about when you'd want to use each one. New users are coming at this with their particular problem in mind, and kind of need an easy decision tree to guide them to the most suitable approach.